control-plane/agents/core/src/volume/mod.rs

```diff
@@ -15,18 +9,11 @@ pub mod specs;
 
 pub(crate) fn configure(builder: common::Service) -> common::Service {
     let registry = builder.get_shared_state::<Registry>().clone();
+    let new_service = service::Service::new(registry);
+    let volume_service = VolumeServer::new(Arc::new(new_service.clone()));
     builder
-        .with_channel(ChannelVs::Volume)
-        .with_default_liveness()
-        .with_shared_state(service::Service::new(registry))
-        .with_subscription(handler!(GetVolumes))
-        .with_subscription(handler!(CreateVolume))
-        .with_subscription(handler!(DestroyVolume))
-        .with_subscription(handler!(ShareVolume))
-        .with_subscription(handler!(UnshareVolume))
-        .with_subscription(handler!(PublishVolume))
-        .with_subscription(handler!(UnpublishVolume))
-        .with_subscription(handler!(SetVolumeReplica))
+        .with_shared_state(new_service)
```

Comment:
Do we need to pass this in? Aren't we just using the VolumeServer?

Reply:
Nope. Removing it.